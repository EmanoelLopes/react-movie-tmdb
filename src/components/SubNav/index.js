import React from 'react';
import { array } from 'prop-types';
import { NavLink } from 'react-router-dom';
import Search from 'components/Search';
import {
  StyledSubNav,
  SubNavList,
  SubNavItem
} from './styles';

const SubNav = ({ items }) => {
  const renderItems = () => {
    return (items.length) && (
      items.map((item, index) => {
        return (
          <SubNavItem key={index}>
            <NavLink
              activeClassName="is-active"
              to={item.link}>
              {item.page}
            </NavLink>
          </SubNavItem>
        );
      })
    );
  };

  return (
    <StyledSubNav>
      <SubNavList>
        {renderItems()}
      </SubNavList>
      <Search />
    </StyledSubNav>
  );
};

SubNav.propTypes = {
  items: array.isRequired,
};

export default SubNav;
