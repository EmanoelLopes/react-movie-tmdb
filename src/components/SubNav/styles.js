import styled from 'styled-components';
import { media, colors } from 'utils/styled';

export const StyledSubNav = styled.nav`
  background-color: ${colors.white};
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-bottom: 25px;
  padding: 25px;

  ${media.desktop`
    flex-direction: row;
  `}
`;

export const SubNavList = styled.ul`
  display: inline-flex;
  justify-content: space-between;

  ${media.desktop`
    justify-content: flex-start;
  `}
`;

export const SubNavItem = styled.li`
  text-align: center;

  ${media.desktop`
    margin-right: 15px;
  `}

  > a {
    color: ${colors.mainColor};
    font-size: 18px;
    text-decoration: none;
  }

  > a:hover {
    text-decoration: underline;
  }

  > a.is-active {
    color: ${colors.movieDBColor};
  }
`;
