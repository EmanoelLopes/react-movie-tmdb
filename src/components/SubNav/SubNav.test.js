import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import SubNav from 'components/SubNav';

const mockItems = [
  { link: '/movies/top-rated', page: 'Top Rated' },
  { link: '/movies/popular', page: 'Popular' },
  { link: '/movies/up-coming', page: 'Up Coming' },
  { link: '/movies/now-playing', page: 'Now Playing' },
];

let wrapped;
let defaultProps = {
  items: [...mockItems]
};

beforeEach(() => {
  wrapped = mount(
    <Router>
      <SubNav {...defaultProps} />
    </Router>
  );
});

afterEach(() => {
  wrapped.unmount();
});

describe('<SubNav />', () => {
  it('should match snapshot', () => {
    expect(wrapped).toMatchSnapshot();
  });
});
