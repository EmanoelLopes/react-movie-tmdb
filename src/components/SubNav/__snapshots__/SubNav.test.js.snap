// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubNav /> should match snapshot 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 25px;
  padding: 25px;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  text-align: center;
}

.c2 > a {
  color: #081c24;
  font-size: 18px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 > a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2 > a.is-active {
  color: #01d277;
}

@media (min-width:62em) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:62em) {
  .c1 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media (min-width:62em) {
  .c2 {
    margin-right: 15px;
  }
}

<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <SubNav
      items={
        Array [
          Object {
            "link": "/movies/top-rated",
            "page": "Top Rated",
          },
          Object {
            "link": "/movies/popular",
            "page": "Popular",
          },
          Object {
            "link": "/movies/up-coming",
            "page": "Up Coming",
          },
          Object {
            "link": "/movies/now-playing",
            "page": "Now Playing",
          },
        ]
      }
    >
      <styled.nav>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": true,
                "lastClassName": "c0",
                "rules": Array [
                  "
  background-color: ",
                  "#fff",
                  ";
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-bottom: 25px;
  padding: 25px;

  ",
                  "
    @media (min-width: ",
                  "62",
                  "em) {
      ",
                  "
    flex-direction: row;
  ",
                  "
    }
  ",
                  "
",
                ],
              },
              "displayName": "styled.nav",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": "nav",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <nav
            className="c0"
          >
            <styled.ul>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-EHOje",
                      "isStatic": true,
                      "lastClassName": "c1",
                      "rules": Array [
                        "
  display: inline-flex;
  justify-content: space-between;

  ",
                        "
    @media (min-width: ",
                        "62",
                        "em) {
      ",
                        "
    justify-content: flex-start;
  ",
                        "
    }
  ",
                        "
",
                      ],
                    },
                    "displayName": "styled.ul",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-EHOje",
                    "target": "ul",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <ul
                  className="c1"
                >
                  <styled.li
                    key="0"
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-bZQynM",
                            "isStatic": true,
                            "lastClassName": "c2",
                            "rules": Array [
                              "
  text-align: center;

  ",
                              "
    @media (min-width: ",
                              "62",
                              "em) {
      ",
                              "
    margin-right: 15px;
  ",
                              "
    }
  ",
                              "

  > a {
    color: ",
                              "#081c24",
                              ";
    font-size: 18px;
    text-decoration: none;
  }

  > a:hover {
    text-decoration: underline;
  }

  > a.is-active {
    color: ",
                              "#01d277",
                              ";
  }
",
                            ],
                          },
                          "displayName": "styled.li",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-bZQynM",
                          "target": "li",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <li
                        className="c2"
                      >
                        <NavLink
                          activeClassName="is-active"
                          aria-current="page"
                          to="/movies/top-rated"
                        >
                          <Route
                            path="\\\\/movies\\\\/top-rated"
                          >
                            <Link
                              aria-current={null}
                              replace={false}
                              to="/movies/top-rated"
                            >
                              <a
                                aria-current={null}
                                href="/movies/top-rated"
                                onClick={[Function]}
                              >
                                Top Rated
                              </a>
                            </Link>
                          </Route>
                        </NavLink>
                      </li>
                    </StyledComponent>
                  </styled.li>
                  <styled.li
                    key="1"
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-bZQynM",
                            "isStatic": true,
                            "lastClassName": "c2",
                            "rules": Array [
                              "
  text-align: center;

  ",
                              "
    @media (min-width: ",
                              "62",
                              "em) {
      ",
                              "
    margin-right: 15px;
  ",
                              "
    }
  ",
                              "

  > a {
    color: ",
                              "#081c24",
                              ";
    font-size: 18px;
    text-decoration: none;
  }

  > a:hover {
    text-decoration: underline;
  }

  > a.is-active {
    color: ",
                              "#01d277",
                              ";
  }
",
                            ],
                          },
                          "displayName": "styled.li",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-bZQynM",
                          "target": "li",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <li
                        className="c2"
                      >
                        <NavLink
                          activeClassName="is-active"
                          aria-current="page"
                          to="/movies/popular"
                        >
                          <Route
                            path="\\\\/movies\\\\/popular"
                          >
                            <Link
                              aria-current={null}
                              replace={false}
                              to="/movies/popular"
                            >
                              <a
                                aria-current={null}
                                href="/movies/popular"
                                onClick={[Function]}
                              >
                                Popular
                              </a>
                            </Link>
                          </Route>
                        </NavLink>
                      </li>
                    </StyledComponent>
                  </styled.li>
                  <styled.li
                    key="2"
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-bZQynM",
                            "isStatic": true,
                            "lastClassName": "c2",
                            "rules": Array [
                              "
  text-align: center;

  ",
                              "
    @media (min-width: ",
                              "62",
                              "em) {
      ",
                              "
    margin-right: 15px;
  ",
                              "
    }
  ",
                              "

  > a {
    color: ",
                              "#081c24",
                              ";
    font-size: 18px;
    text-decoration: none;
  }

  > a:hover {
    text-decoration: underline;
  }

  > a.is-active {
    color: ",
                              "#01d277",
                              ";
  }
",
                            ],
                          },
                          "displayName": "styled.li",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-bZQynM",
                          "target": "li",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <li
                        className="c2"
                      >
                        <NavLink
                          activeClassName="is-active"
                          aria-current="page"
                          to="/movies/up-coming"
                        >
                          <Route
                            path="\\\\/movies\\\\/up-coming"
                          >
                            <Link
                              aria-current={null}
                              replace={false}
                              to="/movies/up-coming"
                            >
                              <a
                                aria-current={null}
                                href="/movies/up-coming"
                                onClick={[Function]}
                              >
                                Up Coming
                              </a>
                            </Link>
                          </Route>
                        </NavLink>
                      </li>
                    </StyledComponent>
                  </styled.li>
                  <styled.li
                    key="3"
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-bZQynM",
                            "isStatic": true,
                            "lastClassName": "c2",
                            "rules": Array [
                              "
  text-align: center;

  ",
                              "
    @media (min-width: ",
                              "62",
                              "em) {
      ",
                              "
    margin-right: 15px;
  ",
                              "
    }
  ",
                              "

  > a {
    color: ",
                              "#081c24",
                              ";
    font-size: 18px;
    text-decoration: none;
  }

  > a:hover {
    text-decoration: underline;
  }

  > a.is-active {
    color: ",
                              "#01d277",
                              ";
  }
",
                            ],
                          },
                          "displayName": "styled.li",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-bZQynM",
                          "target": "li",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <li
                        className="c2"
                      >
                        <NavLink
                          activeClassName="is-active"
                          aria-current="page"
                          to="/movies/now-playing"
                        >
                          <Route
                            path="\\\\/movies\\\\/now-playing"
                          >
                            <Link
                              aria-current={null}
                              replace={false}
                              to="/movies/now-playing"
                            >
                              <a
                                aria-current={null}
                                href="/movies/now-playing"
                                onClick={[Function]}
                              >
                                Now Playing
                              </a>
                            </Link>
                          </Route>
                        </NavLink>
                      </li>
                    </StyledComponent>
                  </styled.li>
                </ul>
              </StyledComponent>
            </styled.ul>
            <Search>
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bdVaJa",
                        "isStatic": true,
                        "lastClassName": "c3",
                        "rules": Array [
                          "
  display: flex;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bdVaJa",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="c3"
                  >
                    <label>
                      Pesquisar: 
                    </label>
                    <input
                      placeholder="Pesquisar títulos..."
                      type="text"
                    />
                  </div>
                </StyledComponent>
              </styled.div>
            </Search>
          </nav>
        </StyledComponent>
      </styled.nav>
    </SubNav>
  </Router>
</BrowserRouter>
`;
